rsocket-core/src/main/java/io/rsocket/internal/ClientServerInputMultiplexer.java

```diff
@@ -79,10 +81,15 @@ public ClientServerInputMultiplexer(DuplexConnection source, PluginRegistry plug
               int streamId = FrameHeaderFlyweight.streamId(frame);
               final Type type;
               if (streamId == 0) {
-                if (isSetup(frame)) {
+                FrameType frameType = FrameHeaderFlyweight.frameType(frame);
+                if (isSetup(frameType)) {
```

Comment:
instead of the isSetup and and isRequestFrame could we just those methods, the if/elses and replace it with switch statement? Seems like we don't have to the extra if/else here if there was one switch statement